packages/flame/lib/src/components/component.dart

```diff
@@ -37,21 +36,46 @@ class Component {
   /// to the root `FlameGame`.
   PositionType positionType = PositionType.game;
 
-  @visibleForTesting
-  LifecycleState get lifecycleState => _state;
-  LifecycleState _state;
+  /// Bitfield which keeps track of the current state of the component: which
+  /// lifecycle events it has already executed, and which are currently being
+  /// executed.
+  ///
+  /// [_initial]: the original state of the component as it was just created. In
+  ///     this state no events has occurred so far.
+  ///
+  /// [_loading]: this flag is set while the component is running its [onLoad]
+  ///     method, and can be checked via [isLoading] getter. More specifically,
+  ///     this bit is set before invoking [onGameResize], it is on for the
+  ///     duration of [onLoad], and then it is turned off when the component is
+  ///     about to be mounted. After that, the bit is never turned on again.
+  ///
+  /// [_loaded]: this flag is set after the component finishes running its
+  ///     [onLoad] method, and can be checked via the [isLoaded] getter. Once
+  ///     set, this bit is never turned off.
+  ///
+  /// [_mounted]: this flag is set when the component becomes properly mounted
+  ///     to the component tree, and then turned off when the component is
+  ///     removed from the tree.
+  ///
+  /// [_removing]: this bit indicates that the component is scheduled for
+  ///     removal at the earliest possible opportunity, and then cleared when
+  ///     the component is actually removed.
```

Comment:
This is an awesome documentation! Great job!

Any chance we can add an example of how this changes when a component is added, changed parent, and then removed?

Reply:
Added more detailed explanation of how state evolves over time.